src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -348,14 +417,361 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     if (edmProperty == null)
                     {
                         ApplyDynamicResourceInNestedProperty(resourceInfoWrapper.NestedResourceInfo.Name, resource,
-                            structuredType, resourceWrapper, readContext);
+                            structuredType, resourceWrapper, nestedReadContext);
+                    }
+                    else
+                    {
+                        ApplyResourceInNestedProperty(edmProperty, resource, resourceWrapper, nestedReadContext);
+                    }
+                }
+            }
+        }
+
+        private static ODataDeserializerContext GenerateNestedReadContext(ODataNestedResourceInfoWrapper resourceInfoWrapper, ODataDeserializerContext readContext, IEdmProperty edmProperty)
+        {
+            ODataDeserializerContext nestedReadContext = null;
+            Routing.ODataPath path = readContext.Path;
```

Comment:
Maybe import the required namespace such that you don't have to use `Routing.ODataPath` repeatedly

Reply:
@gathogojr 
In webapi we use 2 `ODataPath`s
From ODL - `Microsoft.OData.UriParser.ODataPath`
Webapi's - `Microsoft.AspNet.OData.Routing.ODataPath`

`Routing.ODataPath` differentiates the Webapi one from the ODL one.